### PR TITLE
Update gitignore to include Intellij, Mac, and Windows ignores

### PR DESCRIPTION
**Pull request by AustinShalit**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,139 @@
-.idea/
```
Comment (333fred) [inline .gitignore:1]: I'd prefer to keep this, and not the individual .idea ignores below. We don't want any IntelliJ specific stuff in this repo.

Comment (AustinShalit) [inline .gitignore:1]: Ok.  I put it back and removed the specific files.

File: .gitignore [hunk 2]
```diff
@@ -1,10 +1,127 @@
...
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+fabric.properties
+
+### Intellij Patch ###
+# Comment Reason: https://github.com/joeblau/gitignore.io/issues/186#issuecomment-215987721
+
 *.iml
-*~
```
Comment (333fred) [inline .gitignore:3]: Only other thing is that we shouldn't remove this one. It's emacs autobackups.

**Comment by 333fred** [comment]
:+1: modulo comments.

File: .gitignore (hunk 2 above)
Comment (AustinShalit) [inline .gitignore:3]: Ok.  Also added vim ignores while we are at it.

**Pull request merged**